sormas-ui/src/main/java/de/symeda/sormas/ui/symptoms/SymptomsForm.java

```diff
@@ -990,6 +1003,36 @@ private void addListenerForOnsetFields(ComboBox onsetSymptom, DateField onsetDat
 		onsetDateField.setEnabled(false); // will be updated by listener if needed
 	}
 
+	// event for the symptomatic field
+	private void addListenerForOnsetSymptomaticField(DateField onsetDateField) {
+		Field<YesNoUnknown> sourceFieldSymptomatic = (Field<YesNoUnknown>) getFieldGroup().getField(symptomaticFieldId);
+		List<String> allPropertyIds =
+				Stream.concat(unconditionalSymptomFieldIds.stream(), conditionalBleedingSymptomFieldIds.stream()).collect(Collectors.toList());
+		allPropertyIds.add(LESIONS_THAT_ITCH);
+
+		sourceFieldSymptomatic.addValueChangeListener(event -> {
+			for (Object sourcePropertyId : allPropertyIds) {
```

Comment:
The visibility status of the onset date field should only be changed once here, not for every symptom that is processed. Additionally, the onset symptom field should also be taken care of with the same logic as the onset date.

It's probably also a good idea to clear the onset date and onset symptom fields when NO or UNKNOWN are selected because if there's been a value entered before and the field is disabled, users can't remove the value anymore.

Reply:
I also add a clear on onset date when we click on clearAllButton